kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesCrudDispatcher.java

```diff
@@ -75,4 +127,170 @@ public MockResponse handleGet(String path) {
     return response;
   }
 
+  /**
+   * Patches the specified object to the in-memory db.
+   *
+   * @param path
+   * @param s
+   * @return
+   */
+  public MockResponse handlePatch(String path, String s) {
+    MockResponse response = new MockResponse();
+    String body = doGet(path);
+    if (body == null) {
+      response.setResponseCode(404);
+    } else {
+      try {
+        JsonNode patch = context.getMapper().readTree(s);
+        JsonNode source = context.getMapper().readTree(body);
+        JsonNode updated = JsonPatch.apply(patch, source);
+        String updatedAsString = context.getMapper().writeValueAsString(updated);
+
+        AttributeSet attributeSet = null;
+
+        AttributeSet query = attributeExtractor.fromPath(path);
+
+        attributeSet = map.entrySet().stream()
+          .filter(entry -> entry.getKey().matches(query))
+          .findFirst().get().getKey();
```

Comment:
`findFirst()` returns an Optional, you're performing an `Optional#get` with no checks, this will probably break (this has also been reported by Sonar.

Try to chenge that get with `orElse(...)` or `orElseThrow(...)` depending on if we are expecting the key to always be present or not.